Make a GraphqlEmbed MDX component to use in docs.

jean182 opened the pull request:
#24089  Description

In order to address issue [21256](https://github.com/gatsbyjs/gatsby/issues/21256), I added a new component called `GraphqlEmbed` that will allow contributors to use the code sandbox environment and be able to inject the queries in the markdown without writing an ugly iframe snippet, this work rely on the `EggheadEmbed` component and this [PR](https://github.com/gatsbyjs/gatsby/pull/21609), it is pretty similar, just added a few extra props to make it customizable.

### Documentation

Docs files updated in the PR to use the new component.

## Related Issues

  Link to an issue that is partially addressed by this PR (if there are any)
  Addresses issue [21256](https://github.com/gatsbyjs/gatsby/issues/21256)

tesseralis commented on www/src/components/graphql-embed.js:
```diff
@@ -0,0 +1,49 @@
+  url: `https://711808k40x.sse.codesandbox.io/___graphql`,
+}
+
+GraphqlEmbed.propTypes = {
+  lazy: PropTypes.bool,
+  title: PropTypes.string.isRequired,
+  url: PropTypes.string,
+  query: PropTypes.string.isRequired,
+}
```
We're not using PropTypes any more with newer stuff, so you can delete this.

tesseralis commented on www/src/components/graphql-embed.js:
```diff
@@ -0,0 +1,49 @@
+  )
+}
+
+GraphqlEmbed.defaultProps = {
+  lazy: false,
+  url: `https://711808k40x.sse.codesandbox.io/___graphql`,
+}
```
Just set these as default values in the props:
```
export default function GraphqlEmbed({ lazy=false, title, url="…", query }) {
```

tesseralis commented on www/src/gatsby-plugin-theme-ui/components.js:
```diff
@@ -6,13 +6,15 @@ import CloudCallout from "../components/shared/cloud-callout"
   GuideList,
   Pullquote,
   EggheadEmbed,
+  GraphqlEmbed,
```
Since this component is only used in a few docs, I think it's better to import it directly with:

```
import GraphqlEmbed from "@components/graphql-embed"
```

Please remove the changes to this file.

tesseralis commented on www/src/components/graphql-embed.js:
```diff
@@ -0,0 +1,49 @@
+import PropTypes from "prop-types"
+
+export default function GraphqlEmbed({ lazy, title, url, query }) {
+  const ASPECT_RATIO = 2 / 3
```
I'm not sure this is the best approach. In `querying-data`, the width/height are given as 600/300 (for an aspect ratio of  1 / 2). While I'm okay with changing the aspect ratio for consistency, I'd like to leave it up to the docs team (ccing @gillkyle).

jean182 replied:
Sure I can switch the aspect ratio to 1 / 2 to keep consistency.

tesseralis replied:
Well… the problem is some pages have a width/height of 600/400 (for an aspect ratio of 2/3 like you had before). Like I said, I'd be okay with keeping it consistent at 2/3 but I just want approval from the docs team to do so.

jean182 commented:
I addressed the PR comments, I was trying to migrate all the iframes that use the graphql url to use the component but this iframe gave me a lot of linter warnings when I tried to change to use the new component https://github.com/gatsbyjs/gatsby/blob/3ba5d78da7f293ba2a4be3f5ccd6fb275a5ceb91/docs/docs/graphql-reference.md#complete-list-of-possible-operators , for some reason the linter show a lot of warnings
<img width="1440" alt="Screen Shot 2020-07-09 at 03 43 04" src="https://user-images.githubusercontent.com/7577652/87024636-9f4b7880-c196-11ea-99ae-35311852a783.png">
**The warnings disappear if I keep that iframe as it is right now.**
.

tesseralis commented:
@jean182 Ahh, it's because remark incorrectly thinks the query code is markdown and not MDX. I have a PR to add `remark-mdx` at #25662, so if you would wait till that gets merged I think that will fix the problem.

jean182 commented:
> @jean182 Ahh, it's because remark incorrectly thinks the query code is markdown and not MDX. I have a PR to add `remark-mdx` at #25662, so if you would wait till that gets merged I think that will fix the problem.

Sure I can wait, thanks

LekoArts commented:
Hey! Thanks so much for opening this pull request!

Since I closed the `remark-mdx` PR I'll also close this one. Because we've decided that the additional maintenance burden of a new linter (small use cases still) and the benefits of these components over just the iframes are too small.

We absolutely want to have you as a contributor, so please take a look at [our open issues][open-issues] for ideas, and please reach out to us on Twitter at `@gatsbyjs` with questions.

Thanks again, and we look forward to seeing more PRs from you in the future! :muscle::purple_heart:

[open-issues]: https://github.com/gatsbyjs/gatsby/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc
